src/menu/stateful-container.js

```diff
@@ -149,11 +154,71 @@ export default class MenuStatefulContainer extends React.Component<
         }
         this.handleEnterKey(event);
         break;
+      default:
+        if (this.props.typeAhead) {
+          clearTimeout(this.typeAheadTimeOut);
+          this.handleAlphaDown(event);
+        }
+        break;
+    }
+  };
+  handleAlphaDown = (event: KeyboardEvent) => {
+    const rootRef = this.props.rootRef ? this.props.rootRef : this.rootRef;
+    const prevIndex = this.state.highlightedIndex;
+
+    this.typeAheadChars += event.key;
+    this.typeAheadTimeOut = setTimeout(() => {
+      this.typeAheadChars = '';
+    }, 500);
+
+    var nextIndex = prevIndex;
+    event.preventDefault();
+    var list = this.getItems();
+    if (list.length === 0 || !('label' in list[0])) return;
+
+    var notMatch = true;
+    for (let n = 0; n < list.length; n++) {
+      let label = list[n].label;
+      if (
+        label &&
+        label.toUpperCase().indexOf(this.typeAheadChars.toUpperCase()) === 0
+      ) {
+        nextIndex = n;
+        notMatch = false;
+        break;
+      }
+    }
+
+    if (notMatch) {
+      for (let n = 0; n < list.length; n++) {
+        let label = list[n].label;
+        if (
+          label &&
+          label.toUpperCase().indexOf(this.typeAheadChars.toUpperCase()) > 0
+        ) {
+          nextIndex = n;
+          break;
+        }
+      }
+    }
+    this.internalSetState(STATE_CHANGE_TYPES.character, {
+      highlightedIndex: nextIndex,
+    });
+
+    if (this.refList[nextIndex]) {
+      scrollItemIntoView(
+        this.refList[nextIndex].current,
+        // $FlowFixMe
+        rootRef.current,
+        nextIndex === 0,
+        nextIndex === list.length - 1,
+      );
     }
   };
 
   // Handler for arrow keys
   handleArrowKey = (event: KeyboardEvent) => {
+    this.typeAheadChars = '';
```

Comment:
I believe this reset and the similar one in `handleEnterKey` can be removed if the `clearTimeout` call on line 153 was moved from inside the default case up to the top of the `keydown` event handler

Reply:
yes! You are right.